source/common/upstream/load_balancer_impl.h

```diff
@@ -397,14 +400,22 @@ class EdfLoadBalancerBase : public ZoneAwareLoadBalancerBase {
 
   virtual void refresh(uint32_t priority);
 
+  bool noHostsAreInSlowStart();
+
+  virtual void recalculateHostsInSlowStart(const HostVector& hosts_added);
+
   // Seed to allow us to desynchronize load balancers across a fleet. If we don't
   // do this, multiple Envoys that receive an update at the same time (or even
   // multiple load balancers on the same host) will send requests to
   // backends in roughly lock step, causing significant imbalance and potential
   // overload.
   const uint64_t seed_;
 
+  double applyAggressionFactor(double time_factor);
+  double applySlowStartFactor(double host_weight, const Host& host);
+
 private:
+  friend class EdfLoadBalancerBasePeer;
```

Comment:
What's going on here?

Reply:
i use this pattern to assert low level impl details in tests, eg config values for slow start and latest host added time, gives more confidence.

Reply:
somehow all my replies/comments did not persist, so will duplicate them. I use this pattern to assert low level impl details in tests, like values for slow start config options and latest host added time, gives more confidence.